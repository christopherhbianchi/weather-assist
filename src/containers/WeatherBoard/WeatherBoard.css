.weather-board {
  background: #4286f4;
  color: white;
  border-radius: 25px;
  padding-top: 5px;
  padding-bottom: 15px;
}
.weather-card-bumper{
  padding: 15px 15px;
}

button{
  border: none;
  padding: 5px 15px;
  border-radius: 25px;
}
button:hover{
  background-color: #42bcf4;
  color: white;
  transition-duration: 0.4s;
}
