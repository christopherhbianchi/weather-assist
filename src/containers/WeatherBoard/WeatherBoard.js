import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { loadWeather, createWeather, destroyWeatherSuccess, updateWeather } from '../../actions/weatherActions.js';
//import components here
import WeatherCard from '../../components/WeatherCard/WeatherCard.js';
import AddWeather from '../AddWeather/AddWeather.js';
import './WeatherBoard.css'


export class WeatherBoard extends Component {

  //we do this to prevent componentDidMount from loading prior to data being loaded into state
  async componentDidMount(){
    await this.props.loadWeather();//can put it in the redux store now
  }//closes componentDidMount

  removeCard = (weather) => {
    this.props.destroyWeatherSuccess(weather);
  }

  renderCards = () => {
    console.log(this.props.weather);
    return this.props.weather.map( (weather, index) =>
      <div key={index}>
        <WeatherCard weather={weather} removeCard={this.removeCard}/>
        <div className='bumper'></div>
      <div className='weather-card-bumper'></div>
      </div>
    );//closes return
  }//closes renderCards

  render(){
    const weatherArr = this.props.weather;
    console.log('in Weatherboard weatherArr: ' + weatherArr);
    return weatherArr.length < 1 ? null :
      <div className='weather-board'>
        <h3 className='weatherBoardHeader'>Current Weather</h3>
        {
          this.renderCards()
        }
      </div>
  }//closes render
}//closes Component

//This weatherboard is going to need action creators to load weather,
//as well as access to the State, for the state's weather array
const mapStateToProps = (state) => ({
  weather: state.weather
});

const mapDispatchToProps = (dispatch) => ({
  loadWeather: () => dispatch(loadWeather()),
  destroyWeatherSuccess: weather => dispatch(destroyWeatherSuccess(weather)),
  updateWeather: weather => dispatch(updateWeather(weather))
});

WeatherBoard.propTypes = {
  weather: PropTypes.array,
  destroyWeather: PropTypes.func,
  updateWeather: PropTypes.func
};


export default connect(mapStateToProps, mapDispatchToProps)(WeatherBoard);
