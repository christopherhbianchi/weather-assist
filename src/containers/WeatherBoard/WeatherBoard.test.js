import React from 'react';
import { shallow } from 'enzyme';
import { WeatherBoard, mapStateToProps, mapDispatchToProps } from './WeatherBoard.js';


describe('<WeatherBoard />', () => {
  const weatherObj = {
    name:"Los Angeles",
    currently:{
      temperature:70
    }
  };
  const loadWeatherMock = jest.fn();
  const renderedComponent = shallow( <WeatherBoard
    weather={[weatherObj]}
    loadWeather={loadWeatherMock} />
  );

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });//it

  it('should map the store correctly', () =>{
    const mockStore = { weather:[] };
    const mapped = mapStateToProps(mockStore);

    expect(mapped).toEqual(mockStore);
  });//it

  it('should dispatch loadWeather', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.loadWeather();
    expect(mockDispatch).toHaveBeenCalled();
  });//it

  it('should dispatch destroyWeatherSuccess', () => {
    const mockDispatch = jest.fn(); //create a spy
    const mapped = mapDispatchToProps(mockDispatch); //dispatch it to this object

    mapped.destroyWeatherSuccess(); //when this mapped object invokes it
    expect(mockDispatch).toHaveBeenCalled(); //the mockDispatch should have been called
  });//it
});//describe
