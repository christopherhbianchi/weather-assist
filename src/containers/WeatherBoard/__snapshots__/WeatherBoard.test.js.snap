// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherBoard /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherBoard
    loadWeather={[Function]}
    weather={
        Array [
            Object {
              "currently": Object {
                "temperature": 70,
              },
              "name": "Los Angeles",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="weatherBoardHeader"
>
          Current Weather
</h3>,
        Array [
          <div>
            <WeatherCard
                        removeCard={[Function]}
                        weather={
                                    Object {
                                                "currently": Object {
                                                  "temperature": 70,
                                                },
                                                "name": "Los Angeles",
                                              }
                        }
            />
            <div
                        className="bumper"
            />
            <div
                        className="weather-card-bumper"
            />
</div>,
        ],
      ],
      "className": "weather-board",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Current Weather",
          "className": "weatherBoardHeader",
        },
        "ref": null,
        "rendered": "Current Weather",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WeatherCard
              removeCard={[Function]}
              weather={
                            Object {
                                          "currently": Object {
                                            "temperature": 70,
                                          },
                                          "name": "Los Angeles",
                                        }
              }
/>,
            <div
              className="bumper"
/>,
            <div
              className="weather-card-bumper"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "removeCard": [Function],
              "weather": Object {
                "currently": Object {
                  "temperature": 70,
                },
                "name": "Los Angeles",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bumper",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "weather-card-bumper",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="weatherBoardHeader"
>
            Current Weather
</h3>,
          Array [
            <div>
              <WeatherCard
                            removeCard={[Function]}
                            weather={
                                          Object {
                                                        "currently": Object {
                                                          "temperature": 70,
                                                        },
                                                        "name": "Los Angeles",
                                                      }
                            }
              />
              <div
                            className="bumper"
              />
              <div
                            className="weather-card-bumper"
              />
</div>,
          ],
        ],
        "className": "weather-board",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Current Weather",
            "className": "weatherBoardHeader",
          },
          "ref": null,
          "rendered": "Current Weather",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WeatherCard
                removeCard={[Function]}
                weather={
                                Object {
                                                "currently": Object {
                                                  "temperature": 70,
                                                },
                                                "name": "Los Angeles",
                                              }
                }
/>,
              <div
                className="bumper"
/>,
              <div
                className="weather-card-bumper"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "removeCard": [Function],
                "weather": Object {
                  "currently": Object {
                    "temperature": 70,
                  },
                  "name": "Los Angeles",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "bumper",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "weather-card-bumper",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
